crates/tarball/src/lib.rs

```diff
@@ -36,10 +36,10 @@ pub async fn download_and_extract(
         .unwrap();
 
     while let Some(item) = stream.next().await {
-        let chunk =
-            item.or(Err(TarballError::Network("error while downloading file".to_owned())))?;
+        let chunk = item
+            .or_else(|_| Err(TarballError::Network("error while downloading file".to_owned())))?;
```

Comment:
If I understood it correctly, because `or_else` creates a scope (a function) inside, the function (creation of err), is only executed when the error is called. I assumed Rust compiler was smarter than this. Good to know!

Reply:
The compiler is free to optimize things as it sees fit, and so maybe it would be smart enough, but it's better to not rely on smarts if you don't have to :)